addons/sale/migrations/9.0.1.0/post-migration.py

```diff
@@ -55,12 +55,139 @@ def set_crm_team_message_types(env):
             (4, env.ref('sale.mt_salesteam_invoice_created').id)]})
 
 
+@openupgrade.logging()
+def set_so_line_amounts(env):
+    """We replicate here the code of the compute function and set the values
+    finally via SQL for avoiding the trigger of the rest of the computed
+    fields that depends on these fields.
+    """
+    lines = env['sale.order.line'].search([])
+    for line in lines:
+        price = line.price_unit * (1 - (line.discount or 0.0) / 100.0)
+        taxes = line.tax_id.compute_all(
+            price, line.order_id.currency_id, line.product_uom_qty,
+            product=line.product_id, partner=line.order_id.partner_id)
+        env.cr.execute("""
+            UPDATE sale_order_line
+            SET price_tax = %s,
+                price_total = %s,
+                price_subtotal = %s
+            WHERE id = %s""", (
+                taxes['total_included'] - taxes['total_excluded'],
+                taxes['total_included'],
+                taxes['total_excluded'],
+                line.id,
+            )
+        )
+
+
+def set_so_line_computed_rest(env):
+    """Emulate the computation of the rest of the computed fields through
+    equivalente SQL querys.
+    """
+    openupgrade.logged_query(
+        env.cr, """
+        UPDATE sale_order_line
+        SET price_reduce = CASE
+            WHEN abs(product_uom_qty) > 0.00001
+                THEN price_subtotal / product_uom_qty
+            ELSE 0.0
+        END""")
+    openupgrade.logged_query(
+        env.cr, """
+        UPDATE sale_order_line sol
+        SET qty_invoiced = sub.qty
+        FROM (
+            SELECT rel.order_line_id,
+                SUM(CASE
+                    WHEN ai.type = 'out_invoice'
+                        THEN ail.quantity / u.factor * u2.factor
+                    WHEN ai.type = 'out_refund'
+                        THEN -ail.quantity / u.factor * u2.factor
+                    ELSE 0.0
+                END) AS qty
+            FROM sale_order_line_invoice_rel rel
+                JOIN sale_order_line sol2 ON
+                    rel.order_line_id = sol2.id
+                JOIN account_invoice_line ail ON
+                    rel.invoice_line_id = ail.id
+                JOIN account_invoice ai ON ai.id = ail.invoice_id
+                JOIN product_uom u on (u.id = ail.uom_id)
+                JOIN product_uom u2 on (u2.id = sol2.product_uom)
+            WHERE ai.state != ('cancel')
```

Comment:
I know it's the same but feels weird like it is:

```suggestion
            WHERE ai.state != 'cancel'
```